### PR TITLE
Complete and validate

### LINKED ISSUE #1: Implement bag completeness and validness
According to [BagIt spec 16](https://tools.ietf.org/html/draft-kunze-bagit-16#section-3) a bag is **complete** when:

* Every required element is present (`bagit.txt`, `data/`, `manifest-<alg>.txt`)
* Every file listed in every tag manifest is present
* Every file listed in every payload manifest is present
* (BagIt >= 1.0) every payload file is listed in every payload manifest
* Every fetch file is present in the payload

A bag is **valid** when:
* The bag is _complete_
* Every checksum in every payload manifest and tag manifest has been successfully verified against the contents of the corresponding file.

Implement methods `bag.isComplete` and `bag.isValid` according to the statements above.

### PR DESCRIPTION
**Pull request by rvanheest**
Fixes #1

Implements
* [x] `isComplete`
* [x] `isValid`

...according to the BagIt specs. Both methods return `Either[String, Unit]`, meaning that if the bag is complete/valid, it returns `Right[Unit]` and if the bag is not complete/valid, it returns `Left[String]`, with the `String` containing the message why the bag is not complete/valid.

@janvanmansum @lindareijnhoudt @jo-pol for review

### CODE REVIEW

File: src/main/scala/nl/knaw/dans/bag/DansBag.scala [hunk 1]
```diff
@@ -384,6 +384,39 @@ trait DansBag {
    *         `scala.util.Failure` otherwise
    */
   def save(): Try[Unit]
+
+  /**
+   * Verifies if a bag is complete.
+   * According to the BagIt v16 specs, a bag is complete when:
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/bag/DansBag.scala]: use spelled out "version 16" for the release "versions" and "V1.0", "V0.97" for main versions.

File: src/main/scala/nl/knaw/dans/bag/DansBag.scala [hunk 2]
```diff
@@ -384,6 +384,39 @@ trait DansBag {
...
+   * <ul>
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/bag/DansBag.scala]: Use Scaladoc "markdown" (?)

File: src/test/scala/nl/knaw/dans/bag/v0/DansV0BagSpec.scala [hunk 3]
```diff
@@ -2507,4 +2507,206 @@ class DansV0BagSpec extends TestSupportFixture
...
     }
   }
+
+  "isComplete" should "succeed on a complete bag" in {
+    val bag = simpleBagV0()
+
+    (bag / "bagit.txt").toJava should exist
+    bag.data.toJava should exist
+    bag.glob("manifest-*.txt").toList should not be empty
+    bag.fetchFiles shouldBe empty
```
Comment (janvanmansum) [inline src/test/scala/nl/knaw/dans/bag/v0/DansV0BagSpec.scala]: Do you need to check first if your test data is correct?

File: src/main/scala/nl/knaw/dans/bag/DansBag.scala (hunk 2 above)
Comment (rvanheest) [reply src/main/scala/nl/knaw/dans/bag/DansBag.scala]: can't find it. you got a reference?

**Pull request merged**